### PR TITLE
MIR-858 MIR-855 add sword support for ojs and deep green

### PR DESCRIPTION
**Pull request by paschty**
[MIR-855](https://mycore.atlassian.net/browse/MIR-855) and [MIR-858](https://mycore.atlassian.net/browse/MIR-858).

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 4ac7f095aea30ccb6c618ef2987327353b7fd265! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2332-pull_request-412

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 336a61b71aee72bc80e525b7d21a52dbef48fd5e! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2334-pull_request-412

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 5b6aa956ed29a91c4af7887dc17663e66f597f57! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2336-pull_request-412

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 5b6aa956ed29a91c4af7887dc17663e66f597f57! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2336-pull_request-412

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 5b6aa956ed29a91c4af7887dc17663e66f597f57! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2336-pull_request-412

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiIngester.java  9
- mir-module/src/main/java/org/mycore/mir/sword2/MIRSwordIngesterBase.java  2
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSIngester.java  9
- mir-module/src/main/java/org/mycore/mir/sword2/MIRDeepGreepIngester.java  11
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIRDeepGreenCollectionProvider.java  1
         

Clones added
============
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiIngester.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSIngester.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mir/mir/pullRequest?prid=4395605&bid=14900454)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/sword2/MIRDeepGreepIngester.java  11
- mir-module/src/main/java/org/mycore/mir/sword2/MIRDeepGreenCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSIngester.java  9
- mir-module/src/main/java/org/mycore/mir/sword2/MIRSwordIngesterBase.java  2
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiIngester.java  9
         

Clones added
============
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSIngester.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiIngester.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mir/mir/pullRequest?prid=4395605&bid=14900454)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiIngester.java  9
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIRDeepGreepIngester.java  11
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSIngester.java  9
- mir-module/src/main/java/org/mycore/mir/sword2/MIRDeepGreenCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIRSwordIngesterBase.java  2
         

Clones added
============
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiIngester.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSIngester.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mir/mir/pullRequest?prid=4395605&bid=14900454)

File: mir-module/src/main/resources/xsl/sword/jats2mods.xsl [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        </mods:part>
+      </xsl:for-each>
+      <xsl:apply-templates select="issn" />
+      <xsl:call-template name="originInfo" />
+    </mods:relatedItem>
+  </xsl:template>
+  
+  <xsl:template name="originInfo">
+    <xsl:if test="pub-date|publisher">
+      <mods:originInfo>
```
Comment (kkrebs) [inline mir-module/src/main/resources/xsl/sword/jats2mods.xsl]: should be:
`mods:originInfo eventType="publication"`

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/sword2/MIRSwordIngesterBase.java  2
- mir-module/src/main/java/org/mycore/mir/sword2/MIRDeepGreenCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSCollectionProvider.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSIngester.java  9
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiIngester.java  9
- mir-module/src/main/java/org/mycore/mir/sword2/MIRDeepGreepIngester.java  11
         

Clones added
============
- mir-module/src/main/java/org/mycore/mir/sword2/MIROJSIngester.java  1
- mir-module/src/main/java/org/mycore/mir/sword2/MIRGoobiIngester.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mir/mir/pullRequest?prid=4395605&bid=14900454)

**Pull request merged**